mbuild/pattern.py

```diff
@@ -26,8 +26,20 @@ def __len__(self):
     def __getitem__(self, item):
         return self.points[item]
 
-    def scale(self, scalar):
-        self.points *= scalar
+    def scale(self, by):
+        """Scale the points in the Pattern.
+
+        Parameters
+        ----------
+        by : float or np.ndarray, shape=(3,)
+            The factor to scale by. If a scalar, scale all directions isotropically.
+            If np.ndarray, scale each direction independently.
+        """
+        try:
+            for i, dirscale in enumerate(by):
+                self.points[:, i] *= dirscale
```

Comment:
I think you can completely ditch this try/except as long as "by" gets cast to a numpy array first:

```python
In [1]: import numpy as np

In [2]: points = np.array([[1,2,3],[4,5,6]])

In [3]: points
Out[3]: 
array([[1, 2, 3],
       [4, 5, 6]])

In [4]: points * np.array([1, 2, 3])
Out[4]: 
array([[ 1,  4,  9],
       [ 4, 10, 18]])
```